js/bitstamp.js

```diff
@@ -818,8 +856,16 @@ module.exports = class bitstamp extends Exchange {
                         cost = 0.0;
                     cost += trade['cost'];
                     trades.push (trade);
+                    const tradeTimestamp = this.safeInteger (trade, 'timestamp');
+                    if (!lastTradeTimestamp || lastTradeTimestamp < tradeTimestamp)
+                        lastTradeTimestamp = tradeTimestamp;
+                    if (!firstTradeTimestamp || firstTradeTimestamp > tradeTimestamp)
+                        firstTradeTimestamp = tradeTimestamp;
                 }
             }
+            // there is no timestamp from fetchOrder, so the best we have the the first trade time
+            if (!timestamp)
+                timestamp = firstTradeTimestamp;
```

Comment:
I'd rather suggest to set it to `undefined` in this case, because, the `timestamp` should be fixed and should not drift (compared to `lastTradeTimestamp`). What do you say?

Reply:
It would only 'drift' once, from undefined to the firstTradeTimestamp.

But I don't feel strongly - I can remove it no problem.